test/blockstore-test.js

```diff
@@ -77,23 +81,30 @@ describe('BlockStore', function() {
     });
 
     it('has unimplemented base methods', async () => {
-      const methods = ['open', 'close', 'write', 'writeUndo',
-                       'read', 'readUndo', 'prune', 'pruneUndo',
-                       'has', 'hasUndo', 'ensure'];
+      const groups = {
+        base: ['open', 'close', 'ensure'],
+        block: ['write', 'read', 'prune', 'has'],
+        undo: ['writeUndo', 'readUndo', 'pruneUndo', 'hasUndo'],
+        filter: ['writeFilter', 'readFilter', 'readFilterHeader',
+                 'pruneFilter', 'hasFilter'],
+        merkle: ['writeMerkle', 'readMerkle', 'pruneMerkle', 'hasMerkle']
+      };
 
       const store = new AbstractBlockStore();
 
-      for (const method of methods) {
-        assert(store[method]);
+      for (const methods of Object.values(groups)) {
+        for (const method of methods) {
+          assert(store[method]);
 
-        let err = null;
-        try {
-          await store[method]();
-        } catch (e) {
-          err = e;
+          let err = null;
+          try {
```

Comment:
In general, should we be moving these types of tests to `assert.rejects()` ?

Reply:
Yeah, probably. That works there, however the error message would not be customized to show the method name. So it would probably be necessary to make an `it` for each of them, and may be too much.

Reply:
Oh you mean the error message to the tester if the test failed (as in, the method did not throw an error)? Interesting. I wonder if `assert.rejects()` could have a field for an error-message-if-missing-expected-rejection.

Reply:
I don't see anything for that in documentation.

Reply:
Added.